### ISSUE TITLE
Dispatch Workflow - SR Selection (Where are we going)

### ISSUE THREAD

**Issue by trevorskaggs** [trevorskaggs/shelterly#52]
Overall we are going to move towards the Dispatch Workflow being a bit more reliant on the human decision maker at this point.  The Dispatch Workflow will work on 2 primary mechanisms: Filtering and Sorting

The basic flow will be as follows:
- Present a map that allows a user to see the entire extent of all `Open` SRs.  Have the ability to select a smaller area (polygon drawing) that will limit the result set. (Geospatical Filtering)
- Provide buttons that will filter SR list further (Attribute Filtering)
   - ACO Required

Priority Sorting Rules:
- The first sorting priority will be for any SR that has an injured animal.  If the animal also requires special handling from an ACO, the `ACO Required` flag must be set to True for the results to display.
- The second sorting priority will be SRs with the largest number of animals.

The resulting list will be a series of cards that show information about the resulting SRs:
  - At a minimum, it will be important to show the number/type of animals in a given SR.
  - I also think this would be a good opportunity to start developing iconography for the status/filterable traits of the SRs
     - "Turn Around-able" - Perhaps some kind of trailer icon?
     - "Accessible" - Picture of small sedan as that is effectively what we are meaning
     - "ACO Required" - Perhaps some kind of Badge icon would work well here?

Each row will need to have a check box to allow the Dispatch user to select the SR for Evac.  Upon selection of a SR, the remaining list will be sorted by proximity to currently selected SR(s). (Spatial Sorting)

Likely after this step we will want to drop ourselves onto an EA Detail page, for the time being, lets drop ourselves back at the Evac landing page.  We will design this page later in the process as we will likely want this page to be printable.

**Comment by trevorskaggs** [comment]
![image](https://user-images.githubusercontent.com/1276123/87741656-e68ac800-c799-11ea-9bc3-663945c9b9f1.png)

**Comment by AlexMountain** [comment]
Gonna throw out some icons to get feedback.

Here's a car icon already supported by FA: https://fontawesome.com/icons/car?style=solid

For turning around, what came to my mind was something like the U-turn icon.  Something like this (although doesn't look to be natively supported by FA):
![image](https://user-images.githubusercontent.com/5216323/89717957-55b69f00-d989-11ea-9b45-4f8eac936614.png)

I wasn't too impressed with the free badges on FontAwesome...  I dunno, maybe something like this? https://fontawesome.com/icons/shield-alt?style=solid

**Comment by trevorskaggs** [comment]
I was kind of thinking something like the following as really we are indicating if a trailer can navigate the property:
https://fontawesome.com/icons/trailer?style=solid

**Comment by trevorskaggs** [comment]
I like the car on also!

**Comment by trevorskaggs** [comment]
Badge option?
https://fontawesome.com/icons/badge-sheriff?style=light

**Comment by AlexMountain** [comment]
I will go ahead and add the car and trailer respectively.  The badge you linked is a "pro" icon that we would have to pay for.  Also I like the solid badge more, seems cleaner, however, is still pro: https://fontawesome.com/icons/badge-sheriff?style=solid

@trevorskaggs Can you (or someone you assign) investigate the best way to going about creating a custom icon for the badge?  Finding an alternative source would also probably work here?

**Comment by AlexMountain** [comment]
Injured animals seems to be an important indicator if we're sorting with it as priority 1, should we add an icon to the SR card so that users can tell at a glance?  Maybe something like https://fontawesome.com/icons/band-aid?style=solid ?

**Comment by AlexMountain** [comment]
What if instead of providing an area selection box to limit the result set, we make it dynamic based on whatever is currently visible on the map?  The user could move around and adjust the zoom to match their desired window size.  I feel like this is how some of the modern map pickers work, Yelp, DoorDash, GoogleMaps, etc.  I feel like it also achieves the desired functionality more intuitively and with fewer steps.  Thoughts?

**Comment by AlexMountain** [comment]
Note in case we want to adjust:  Cats don't have a medium size as depicted in the sample.  Cat sizes currently:

```
const catSizeChoices = [
  {value:'small', label:'Small'},
  {value:'full', label:'Full Size'},
]
```

Just making sure we want it to say "2 Full Size Cats"?

**Comment by trevorskaggs** [comment]
- I like the Bandaid and agree, similar to ACO Required, we should be calling these things out.
- If the cost of doing it this way is fairly low we can give it a try.  What we are trying to avoid is selecting SRs that are in the fire and they dont have a consistent shape.
-I think we are good on the Cats, its basically kittens vs cats.

**Comment by AlexMountain** [comment]
Can we trim down words here by maybe using "Adult Cat" instead?  I realize I'm splitting hairs but it will save some space!  Full Size cats, while technically correct, also sounds funny, I dunno!

Cost turned out to be pretty low, I think I got show/hide working based on the current map zoom/view.  I still need to do some refactoring before this is truly ready for testing, but I think I got all of the features we wanted on this first pass.  Outstanding:

- [ ] Spatial sorting
- [ ] Submitting selection (essentially need EA model first)
- [ ] Proximity heat map?  radius circles?
- [ ] Custom badge icon?

It should be functional if you wanted to pull it down for a feature test to make sure I got em all.
